src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `The side of the figure cannot be less than or equal to zero. Invalid values: a = ${a}, b = ${b}, c = ${c}`,
+      );
+    }
```

Comment:
The condition checks if any side of the triangle is less than or equal to zero, which is correct. However, ensure that the error message is clear and concise, as it currently repeats the same information twice.